doc/specifications/proposed/named_queries/developer_documentation.md

```diff
@@ -0,0 +1,202 @@
+The `ez_query` actions support template override. They support several matchers.
+- `QueryType\Name`: matches the QueryType's name
+- `QueryType\Parameter`: matches the value of one or several of the parameters passed to the QueryType
+- `QueryType\Expression`: matches the parameters with an [Expression](http://symfony.com/en/doc/current/components/expression_language/index.html).
+
+Assuming we want to render the `AcmeBundle:LatestContent` query using
+`AcmeBundle:query/list/latest_articles.html.twig` when the `ContentType` parameter has the value 'article':
+
+```twig
+system:
+    default:
+        # Or content_query to create a template override for a content query
+        location_query:
+            # Matches the ViewType passed to the controller
+            list:
+                latest_articles:
+                    template: "AcmeBundle:query/list/latest_articles.html.twig"
+                    match:
+                        QueryType\Name: 'AcmeBundle:LatestContent'
+                        QueryType\Parameters: {ContentType: "article"}
+```
+
+#### Template matchers
+
+##### `QueryType\Name`
+Matches the QueryType's name (exact match).
+
+##### `QueryType\Parameters`
+Matches the parameters against a given hash. The hash must contain the parameter's name as the key,
+and the match value as the value:
+
+```yaml
+match:
+  QueryType\Parameters:
+    type: "article",
+    category: "development"
+```
+
+Each parameter will be matched exactly.
+The matcher will match if ALL of the provided parameters match what is contained in the hash.
+
+##### `QueryType\Expression`
+Uses Symfony's [Expression Language](http://symfony.com/en/doc/current/components/expression_language/index.html)
+for advanced matching.
+This matcher expects a valid expression language string as the input. The expression must be
+evaluated to a boolean. The parameters hash is available as `parameters`
+
+In the example below, 'type' must be either 'article' or 'blog_post'.
+
+```yaml
+match:
+  QueryType\Expression: "parameters['type'] in ['article', 'blog_post']"
+```
+
+### Query controller templates
+The templates used by this controller provide you with the results from the query. The results can be iterated over, and
+displayed using the object's properties or the `ez_content` controller actions.
+
+#### Available variables
+
+`location_list`/`content_list` | array | Array of search results, depending if the action (locationQuery or contentQuery)
+`list_count`                   | int   | Number of items in the resultset, within the limit if any
+`total_count`                  | int   | Total number of items in the search result
+
+## QueryType objects
+To make a new QueryType available to the Query Controller, you need to create a class that implements the QueryType
+interface, and register it as such in the service container.
+
+The interface has two methods:
+- `getName(): string`
+  Returns the QueryType name.
+- `getQuery( array $parameters = null ): Query`
```

Comment:
I'd prefer `array()` as default value instead of null. Would avoid further checks.